scripts/create-release.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+cd "$SCRIPT_DIR/.."
+
+version=${1:-}
+if [[ -z "$version" ]]; then
+  echo "USAGE: $0 <version>" >&2
+  exit 1
+fi
+
+# Check if the version matches the semver pattern (without suffixes)
+semver_regex="^([0-9]+)\.([0-9]+)\.([0-9]+)$"
+if [[ ! "$version" =~ $semver_regex ]]; then
+  echo "Version must match the semver pattern (e.g., 1.0.0, 2.3.4)" >&2
+  exit 1
+fi
+
+if [[ "$(git symbolic-ref --short HEAD)" != "master" ]]; then
+  echo "must be on master branch" >&2
+  exit 1
+fi
+
+# Ensure there are no uncommitted or unpushed changes
+uncommited_changes=$(git diff --compact-summary)
+if [[ -n "$uncommited_changes" ]]; then
+  echo -e "There are uncommited changes, exiting:\n${uncommited_changes}" >&2
+  exit 1
+fi
+git pull git@github.com:nix-community/disko master
+unpushed_commits=$(git log --format=oneline origin/master..master)
+if [[ "$unpushed_commits" != "" ]]; then
+  echo -e "\nThere are unpushed changes, exiting:\n$unpushed_commits" >&2
+  exit 1
+fi
+
+previous_version=$(nix --extra-experimental-features nix-command eval --raw -f version.nix version)
+
+# Update the version file
+echo "{ version = \"$version\"; released = true; }" > version.nix
+
+# Update all references to the previous version in the docs
+sed -i "s|v${previous_version}|v${version}|g" **/*.md
+
+# Commit and tag the release
+git commit -am "release: v$version"
```

Comment:
Ideally also run `nix-fast-build` here to make sure that all tests passes.
It delays the release process a bit, but makes sure we don't deliver broken releases.

Reply:
Hmm yes, that makes sense, but I think I need a bit of guidance here. What exactly should `nix-fast-build` build, and with what parameters? I tried to run without `-j` and `--eval-workers`, but that clogged my system so hard I had to reset it. I'm on an i7-4790K, not the newest CPU, but it has 4 cores, 8 threads.

Only `-j 3` and `--eval-workers 3` allowed me to still use the system while the build was happening, but hardcoding these values seems unwise. I'm also unsure if I should just build all `.#checks` or anything else as well.

Reply:
Just go with --eval-workers 3 for now. We only need to run it on a limited set of maintainer machines.

Reply:
Even 1 would be acceptable, because it than just prevents unbound memory usage.

Reply:
Alright, done. I tried it out a bit more, and it seems because the worker RAM limit is at 4GB, only two workers is the safest while still just taking 25 minutes for a full build. It seems like maybe it's possible to use more builder threads and less eval workers, eval seems very memory intensive (and potentially leaky), but it's good enough for now.